DOCSUP-1675 Docs for the cast_keep_nullable setting (#14508)
ClickHouse/ClickHouse: docs/en/operations/settings/settings.md

Default value: `120`.

## cast_keep_nullable {#cast_keep_nullable}

> Enables or disables converstion to nullable types for [CAST](../../sql-reference/functions/type-conversion-functions.md#type_conversion_function-cast) operator. 

Reviewer: ```suggestion
Enables or disables keeping of the `Nullable` data type in [CAST](../../sql-reference/functions/type-conversion-functions.md#type_conversion_function-cast) operations.
```

Reviewer: Also we need a link to Nullable.

If set, `CAST(something_nullable AS Type)` returns `Nullable(Type)`.

Possible values:

> -   0 — Nullable type converstion disabled.
> -   1 — Nullable type converstion enabled.

Reviewer: ```suggestion
-  0 — The final type of `CAST` exactly the destination data type specified.
-  1 — The final type of `CAST` becomes `Nullable(DestinationDataType).
```

Default value: `0`.

**Examples** 

> Query with nullable type converstion disabled:

Reviewer: ```suggestion
The following query exactly results in the destination data type:
```

```sql
SET cast_keep_nullable = 0;
SELECT CAST(toNullable(toInt32(0)) AS Int32) as x, toTypeName(x);
```

Result:

```text
┌─x─┬─toTypeName(CAST(toNullable(toInt32(0)), 'Int32'))─┐
│ 0 │ Int32                                             │
└───┴───────────────────────────────────────────────────┘
```

> Query with nullable type converstion enabled:

Reviewer: ```suggestion
The following query results in the `Nullable` modification on the destination data type:
```
